### PR TITLE
Spectrum Analyzer: Use integer values when working with sample rates

### PR DESCRIPTION
**Pull request by dNechita**
This way we avoid comparing floating point values which is not a reliable thing to do anyway.
This fixes an issue that happens on OpenSuse Leap 42.3 where such a float comparison fails (61.44 != 61.39999).

Signed-off-by: Dan Nechita <dan.nechita@analog.com>

### CODE REVIEW

File: plugins/spectrum_analyzer.c [hunk 1]
```diff
@@ -159,29 +159,29 @@ static ssize_t demux_sample(const struct iio_channel *chn,
...
+		iio_channel_attr_write_longlong(ch0, "sampling_frequency", freq_hz);
 	else
 		fprintf(stderr, "Failed to retrieve iio channel in %s\n", __func__);
 }
 
-static double device_get_rx_sampling_freq(struct iio_device *dev)
+static long long device_get_rx_sampling_freq(struct iio_device *dev)
 {
-	double freq = 0.0;
+	long long freq_hz = 0.0;
```
Comment (commodo) [inline plugins/spectrum_analyzer.c]: `long long freq_hz = 0;` ?

File: plugins/spectrum_analyzer.c [hunk 2]
```diff
@@ -398,11 +405,6 @@ static bool configure_data_capture(plugin_setup *setup)
 		dev_info->adc_scale = '?';
 		dev_info->adc_freq = 0.0;
```
Comment (commodo) [inline plugins/spectrum_analyzer.c]: maybe also change this to `dev_info->adc_freq = 0;`
no idea if some compiler warnings could trigger

File: plugins/spectrum_analyzer.c (hunk 1 above)
Comment (dNechita) [reply plugins/spectrum_analyzer.c]: yes

File: plugins/spectrum_analyzer.c (hunk 2 above)
Comment (commodo) [reply plugins/spectrum_analyzer.c]: oh; wait; `adc_freq` is still double

Comment (dNechita) [reply plugins/spectrum_analyzer.c]: `dev_info->adc_freq` remains of type float so no actions needed here

**Pull request merged**